### ISSUE TITLE
TypeCastExpression could not be resolved

### ISSUE THREAD

**Issue by 7yp0** [evcohen/jsx-ast-utils#66]
Hi,

I was just upgrading eslint and all its plugins and now getting this error when trying to lint with the following setup:
- "eslint": "4.16.0",
- "eslint-config-airbnb": "16.1.0",
- "eslint-plugin-flowtype": "2.41.0",
- "eslint-plugin-import": "2.8.0",
- "eslint-plugin-jsx-a11y": "6.0.3",
- "eslint-plugin-mocha": "4.11.0",
- "eslint-plugin-react": "7.5.1",
- "babel-eslint": "8.2.1",

Full Error Message:
```
Error: The prop value with an expression type of TypeCastExpression could not be resolved.
  Please file issue to get this fixed immediately.
    at Object.extract [as JSXExpressionContainer] (/Users/mil/Development/wizard/node_modules/jsx-ast-utils/lib/values/expressions/index.js:179:11)
    at getValue (/Users/mil/Development/wizard/node_modules/jsx-ast-utils/lib/values/index.js:48:27)
    at extractValue (/Users/mil/Development/wizard/node_modules/jsx-ast-utils/lib/getPropValue.js:24:12)
    at getPropValue (/Users/mil/Development/wizard/node_modules/jsx-ast-utils/lib/getPropValue.js:41:10)
    at /Users/mil/Development/wizard/node_modules/eslint-plugin-jsx-a11y/lib/rules/no-static-element-interactions.js:78:95
    at Array.some (<anonymous>)
    at JSXOpeningElement (/Users/mil/Development/wizard/node_modules/eslint-plugin-jsx-a11y/lib/rules/no-static-element-interactions.js:77:52)
    at listeners.(anonymous function).forEach.listener (/Users/mil/Development/wizard/node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/mil/Development/wizard/node_modules/eslint/lib/util/safe-emitter.js:47:38)
```

Any idea?

**Comment by ljharb** [comment]
What's the code that triggers this?

**Comment by 7yp0** [comment]
I was able to narrow it down to this bit:
```
<NavListItem
  data-tracking-button="context-share"
  title={this.shareLabel}
  iconRight="sharing"
  onClick={(this.handleSharingOverlayShow: any)}
  divider
/>
```
so basically a type cast to any in an onClick.   

I was able to fix this by doing the type cast beforehand.
`const handleSharingOverlayShow: any = this.handleSharingOverlayShow;`   

Thanks for the pointer!  
I didn't know, that this was caused by something in our code.   

Would be nice to have a more detailed error message, or eslint rule for this though.   

I will close this, thanks :) 👍

**Comment by ljharb** [comment]
I think this should be reopened; the rule shouldn’t crash on that.

**Comment by 7yp0** [comment]
Good Point

**Comment by jessebeach** [comment]
Found it. I'm adding a test case and getting this fixed.

**Comment by jessebeach** [comment]
@7yp0 , Fixed in version v2.2.1. This will log an error in the console, but it will no longer throw.
